### PR TITLE
Remove environment variable dependency for mock auth server

### PR DESCRIPTION
**Pull request by teneightfive**
### What changed

This change removes the extra bit of configuration and simplifies the
implementation of the test suite.

### Why did it change

Currently the mock auth server relies on environment variables being
set to define the username and password.

This is a legacy implementation from when we used to hit the real
auth service to run end-to-end tests.

Since mocking has been introduced there is no reason to keep these
values in environment variables anymore.

#### Automated testing

- [ ] Added unit tests to cover changes
- [ ] Added end-to-end tests to cover any journey changes

#### Manual testing

Has been tested in the following browsers:

- [x] Chrome
- [x] Firefox
- [ ] Edge
- [ ] Internet Explorer

### Environment variables

- [ ] Documented in the [README](ministryofjustice/hmpps-book-secure-move-frontend/blob/master/README.md)
- [ ] Added to [continous integration](https://app.circleci.com/pipelines/github/ministryofjustice/hmpps-book-secure-move-frontend)
- [ ] Added to staging environment (deployment repository)
- [ ] Added to preproduction environment (deployment repository)
- [ ] Added to production environment (deployment repository)

### Other considerations

Commit messages with a `fix` or `feat` type are automatically used to generate the [changelog](ministryofjustice/hmpps-book-secure-move-frontend/blob/master/CHANGELOG.md) and
[GitHub release notes](https://github.com/ministryofjustice/hmpps-book-secure-move-frontend/releases) during the release task. Please make sure they will read well on their own in a
summary of changes and that the commit body gives a more detailed description of those changes if necessary.

- [x] No new [Personally Identifiable Information (PII)](https://support.google.com/analytics/answer/6366371?hl=en) is being sent via analytics
- [x] Update [README](ministryofjustice/hmpps-book-secure-move-frontend/blob/master/README.md) with any new instructions or tasks

### CODE REVIEW

File: config/index.js [hunk 1]
```diff
@@ -189,42 +189,70 @@ module.exports = {
...
-      },
+    USERS: {
       COURT: {
-        username: process.env.E2E_COURT_USERNAME,
-        password: process.env.E2E_COURT_PASSWORD,
+        locations: ['ABDRCT', 'BDFRCT'],
+        name: 'End-to-end Court',
+        authorities: ['ROLE_PECS_COURT'],
+        username: 'E2E_COURT',
+        password: 'ZZDBm6Bf',
```
Comment (merlinc) [inline config/index.js:198]: If we are steering into have completely different local passwords to the ones in environments, can we make them memorable? This way when we're running locally we won't have to scroll to line `194`-`255` of a `config` file to figure out the correct password.

File: mocks/auth-server.js [hunk 2]
```diff
@@ -124,15 +70,10 @@ app.get('/auth/logout', (req, res) => {
   res.redirect(req.query.redirect_uri)
 })
 
-const getUserFromToken = req => {
```
Comment (merlinc) [inline mocks/auth-server.js:127]: Function hasn't been deleted, Github has just misplaced the move. :+1:

File: mocks/auth-server.js [hunk 3]
```diff
@@ -3,76 +3,22 @@
...
 
 const { decodeAccessToken } = require('../common/lib/access-token')
 const {
-  E2E: { ROLES },
+  E2E: { USERS },
 } = require('../config')
 
-const roleConfig = {
-  COURT: {
-    locations: ['ABDRCT', 'BDFRCT'],
```
Comment (merlinc) [inline mocks/auth-server.js:14]: We're already pulling in config from the main app (see line `9`) so there's not much difference in where this config lives.

I'm not trilled about having test config in the main app config, but I'm not seeing a cleaner way of dealing with this at the moment.

Comment (teneightfive) [reply mocks/auth-server.js:14]: Yeah, good point. I did think of moving this just to the auth server but some of the config is pulled in in various places in the e2e tests. So config still seemed the most logic place...

File: config/index.js (hunk 1 above)
Comment (teneightfive) [reply config/index.js:198]: Yep, not against that. Could just use a simple format seeing as it's only for mock test purposes anyway.

**Review by merlinc — APPROVED** [review]

**Pull request merged**